gradio/processing_utils.py

```diff
@@ -271,7 +272,19 @@ def save_file_to_cache(file_path: str | Path, cache_dir: str) -> str:
     return full_temp_file_path
 
 
-def check_public_url(url: str):
+def resolve_with_google_dns(hostname: str) -> str | None:
+    url = f"https://dns.google/resolve?name={hostname}&type=A"
```

Comment:
It was hard to find concrete numbers, but it looks like the rate limit for this API is 100 queries/second: https://groups.google.com/g/public-dns-discuss/c/p1o62SJElck, which I'm worried we'll hit from Spaces infra.

E.g. imagine a single popular Space that has a `gr.Examples` that includes a file from a Hugging Face dataset, now multiplied across multiple viral Spaces that share the same IP address